Backend otf

mhoffman opened the pull request:
Hi,

 could you try and see if **your** OTF backend still works for you as before after merging these couple of changes? You'll see that mostly evolved the *.ini file format to support **your** bystanders to completely get rid of `lxml` as a hard dependency.

I now understand why you expect the explicit '\\n' in the `otf_rate` because the lxml parser strips out any '\n'. However it looks like the *.ini format would solve this problem better. That way one can simply enter multiline expressions with triple quotes. Can you confirm that?

Best,
Max.

mhoffman committed:
Followed up with lesss verbose travis unit test

mhoffman committed:
Keep kmos.types from failing is lxml not available

mhoffman committed:
Made kmos.types.Model.__repr__ fall back on .ini format

If lxml is not available

mhoffman committed:
Updated release notes

mhoffman committed:
Added unit test for format conversion between xml and ini format

mhoffman committed:
Fixed formatting of get_ase_constructor a little

mhoffman committed:
Travis CI caching ...

mhoffman committed:
Included filter for site names for kmos.types.Lattice.generate_coord_set

mhoffman committed:
Expanded generate coord set method if several sites are present with same prefix

mhoffman committed:
Merge branch 'develop' into backend-otf

mhoffman committed:
Updated reference test files

mhoffman committed:
Adapter *.ini file format to support bystander elements

mhoffman committed:
Updated reference test files

mhoffman committed:
Autopepped kmos/types.py

mhoffman committed:
Addded one remark to OTF documentation

mhoffman committed:
Added missing default for OTF rate in *ini file reader for

mhoffman committed:
Changed line-splitting in otf-rate parser using builtin re module to be more flexible

mhoffman committed:
Merge remote-tracking branch 'jmlorenzi/backend-otf' into backend-otf

jmlorenzi commented:
Hi Max,

Thanks a lot for this. I am right now actually trying to make some TravisCI-like test for the otf exporter. Once I finish this (most likely today) I will merge this and see what the tests tell me.

jmlorenzi commented on tests/test_types/reference/AB_model.ini:
```diff
@@ -0,0 +1,144 @@
+
+[Process AB_react_down]
+rate_constant = 1/(beta*h)*exp(-beta*E_react*eV)
+otf_rate = None
```
This is being interpreted as a `"None"` string by the importer... I will fix this, to catch the string and pass as a built-in `None`.
May be we should also explcitely prohibit variables called `"None"` across all the code?
Related: Behavior up to know has been to not print anything in the xml file when `process.otf_rate` evaluates to false. I did this to ensure backwards-compatibility (i.e. continue exporting the same .xml when you do not define otf_rate). I do not see a big compelling reason to change this right now.

mhoffman committed:
Updated test data to include otf_rate default

mhoffman committed:
Add nosetests in parallel by default

mhoffman committed:
Added nosetest for compiling model and comparing the first 10000 steps

mhoffman committed:
Disable parallel test execution because it seems to trip up f2py

mhoffman committed:
Removed JANAF dependency from unit test; updated reference files

mhoffman committed:
Put gfortran installation into travis setup

mhoffman committed:
Merge branch 'develop' into backend-otf

mhoffman committed:
Fixed some more typos

mhoffman committed:
Added forgotten link about bystander flag

mhoffman committed:
Added function to shorten variable names on source code export

Defaut length is 15, can be changed with "kmos -l <N> export .."

mhoffman committed:
Changed default variable length to 95 for unbreaking tests for now

mhoffman committed:
Quickfix for get_rate(proc_nr) in OTF base version

mhoffman committed:
Adjusted base.get_rate & model.print_accum_rate_summation to work with OTF backend

mhoffman committed:
Fixed interpretation of otf_rate = None when reading in *.ini files

mhoffman committed:
Update test reference file

mhoffman committed:
Added getter function for inspecting user parameters

mhoffman committed:
Fixed coordinates in tooltip for empty sites in gui editor

mhoffman committed:
Merge branch 'develop' into backend-otf

mhoffman committed:
Added __cmp__ magic method for coord order to make order unique

mhoffman committed:
Take bystanders into account to calculating optimal zoom level

mhoffman committed:
Changed get_std_sampled_data slightly, to samples always the total length (i.e. computational cost) and samples only changes how many subsamples are taken for coverages

mhoffman committed:
Merge branch 'develop' into backend-otf

mhoffman committed:
Added a progressbar for sampling run

mhoffman committed:
Added integrated time to std sampled output  (no just incremental time) for plotting time series;
Added option for side-loading catmap model into "kmos shell" (for using both together in a translated scenario

mhoffman committed:
Merged from develop

mhoffman committed:
Fixed ConfigParser to be case-sensitive on option names

mhoffman committed:
Fixed *.ini file writer to also preserve case on export

mhoffman committed:
Set min/max defaults of Parameters to 0. instead of None to be more consistent with the *.xml file reader

mhoffman committed:
*.ini file reader set default value of tof_count to {} instead of "" to not crash the runtime

mhoffman committed:
Merged from develop

mhoffman committed:
Added nice settings for povray output from Michal Bajdich to export movie function

mhoffman committed:
Added option for string and dict output of get_std_sampled_data

mhoffman committed:
Fixed typo

mhoffman committed:
Clarified error message

mhoffman committed:
Added initial version of initial state detection based on Rossetti et al.

mhoffman committed:
Merge branch 'develop' into backend-otf

mhoffman committed:
Adapted AB model script for ini file format

mhoffman committed:
Moved gazpacho hosting from debian to gist.github.com

mhoffman committed:
Added in movie export to .traj files

mhoffman committed:
Let updated pip first

mhoffman committed:
Try installing ase instead of python-ase

mhoffman committed:
Updated .travis.yml

mhoffman committed:
Catch import error if ase.io.png is not available

mhoffman committed:
Merge branch 'develop' of https://github.com/mhoffman/kmos into develop

mhoffman committed:
Fixed error message for compile failure

mhoffman committed:
Updated traj pickle file to current ASE format

mhoffman committed:
Merged from develop

mhoffman committed:
Made progress bar optional and turn it OFF by default in kmos.run.get_std_sampled_data

mhoffman committed:
Merge branch 'backend-otf' of https://github.com/mhoffman/kmos into backend-otf
